torch/nested/_internal/ops.py

```diff
@@ -1260,6 +1260,34 @@ def _nested_get_max_seqlen(func, *args, **kwargs):
     inp = new_kwargs.pop("input")
     return inp._metadata_cache.get("max_seqlen", None)
 
+@register_jagged_func(torch.ops.aten.masked_select.default, "self: jt, mask: jt")
+def masked_select_default(func, *args, **kwargs):
+    _, new_kwargs = normalize_function(
+        func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
+    )
+    inp = new_kwargs.pop("input")
+    mask = new_kwargs.pop("mask")
+
+    mask_values = None
+    if mask._offsets.equal(inp._offsets):
+        mask_values = mask.values()
+    elif len(mask._values) == len(inp._offsets) - 1:
+        mask_values = torch.repeat_interleave(
+            mask.values(),
+            (inp._lengths if inp._lengths is not None else torch.diff(inp._offsets)),
+        )
+    else:
+        raise RuntimeError(
+            f"Jagged mask with offsets {mask.offsets()} is not compatible with input's offsets {inp.offsets()}"
+        )
+
+    res_values = inp._values.masked_select(mask_values)
+    mask_cumsum = torch.cat(
+        [torch.tensor([0], device=mask.device), mask_values.cumsum(dim=0)]
+    )
+    res_offsets = mask_cumsum[inp._offsets]
+
+    return NestedTensor(values=res_values, offsets=res_offsets)
```

Comment:
```suggestion
    return NestedTensor(values, offsets, **extract_kwargs(indices))
```
Use extract_kwargs to propagate any cached metadata that might have been saved on the input tensor.

Reply:
Since I'm providing offsets I'll manually add metadata_cache and ragged_idx